src/librustc_trans/back/link.rs

```diff
@@ -676,8 +679,12 @@ fn link_natively(sess: &Session,
     } else {
         &sess.target.target.options.pre_link_objects_dll
     };
-    for obj in pre_link_objects {
-        cmd.arg(root.join(obj));
+    // When dynamically linking to MUSL we don't need to pass our startup
+    // objects as those will be provided by the MUSL toolchain
+    if !musl || crt_static {
```

Comment:
Yeah I think all options are configurable from json files now (we round-trip through that format to read our own options IIRC).

I mean all the options in the custom target specs are *already* super specific... I guess I wouldn't really have a problem with those.